scipy/interpolate/_bsplines.py

```diff
@@ -445,34 +476,87 @@ def integrate(self, a, b, extrapolate=None):
-        x = np.asarray([a, b], dtype=np.float_)
-        out = np.empty((2, prod(c.shape[1:])), dtype=c.dtype)
-        _bspl.evaluate_spline(t, c.reshape(c.shape[0], -1),
-                k, x, 0, extrapolate, out)
-        out = out[1] - out[0]
-        return out.reshape(c.shape[1:])
+        if extrapolate == 'periodic':
+            # Split the integral into the part over period (can be several
+            # of them) and the remaining part.
+
+            ts, te = self.t[k], self.t[n]
+            period = te - ts
+            interval = b - a
+            n_periods, left = divmod(interval, period)
+
+            if n_periods > 0:
+                # Evaluate the difference of antiderivatives.
+                x = np.asarray([ts, te], dtype=np.float_)
+                _bspl.evaluate_spline(t, c.reshape(c.shape[0], -1),
+                                      k, x, 0, False, out)
+                integral = out[1] - out[0]
+                integral *= n_periods
+            else:
+                integral.fill(0)
+
+            # Map a to [ts, te], b is always a + left.
+            a = ts + (a - ts) % period
+            b = a + left
+
+            # If b <= te then we need to integrate over [a, b], otherwise
+            # over [a, te] and from xs to what is remained.
+            remainder_int = np.empty_like(out)
+            if b <= te:
+                x = np.asarray([a, b], dtype=np.float_)
+                _bspl.evaluate_spline(t, c.reshape(c.shape[0], -1),
+                                      k, x, 0, False, remainder_int)
+                integral += remainder_int[1] - remainder_int[0]
+            else:
+                x = np.asarray([a, te], dtype=np.float_)
+                _bspl.evaluate_spline(t, c.reshape(c.shape[0], -1),
+                                      k, x, 0, False, remainder_int)
+                integral += remainder_int[1] - remainder_int[0]
+
+                x = np.asarray([ts, ts + b - te], dtype=np.float_)
+                _bspl.evaluate_spline(t, c.reshape(c.shape[0], -1),
+                                      k, x, 0, False, remainder_int)
+                integral += remainder_int[1] - remainder_int[0]
+        else:
+            if not extrapolate:
+                # Shrink the integration interval, if needed.
+                a = max(a, self.t[self.k])
+                b = min(b, self.t[-self.k - 1])
+
+                if self.c.ndim == 1:
+                    # Fast path: use FITPACK's routine
+                    # (cf _fitpack_impl.splint).
+                    t, c, k = self.tck
+                    aint, wrk = _dierckx._splint(t, c, k, a, b)
+                    return aint
+
+            # Evaluate the difference of antiderivatives.
+            x = np.asarray([a, b], dtype=np.float_)
+            _bspl.evaluate_spline(t, c.reshape(c.shape[0], -1),
+                                  k, x, 0, extrapolate, out)
+            integral = out[1] - out[0]
+
+        integral *= sign
```

Comment:
I have done it here @nmayorov  , after the end of both the `if` and `else` loop.